common/src/main/java/io/pravega/common/util/RequestIdGenerator.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.util;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * This is used to generated unique request ids per process.
+ */
+public class RequestIdGenerator {
```

Comment:
Two different requestId generators will likely produce collisions.

Reply:
The idea here was to have RequestIds unique per client process. Having the same request id across different client process would not cause any issues on the PRP (or client) as the reply is always sent on the connection which sent the request.
(Please let me know if I misunderstood your comment).